src/pages/allSearchResults/index.jsx

```diff
@@ -0,0 +1,115 @@
+import EllipsisIcon from '../../assets/icons/ellipsisIcon'
+import { useState, useEffect, useRef } from 'react';
+import {useLocation, Link } from 'react-router-dom';
+import SearchIcon from '../../assets/icons/searchIcon';
+import ArrowLeftIcon from '../../assets/icons/arrowLeftIcon';
+import ProfileCircle from '../../components/profileCircle';
+import Header from '../../components/header';
+import Navigation from '../../components/navigation';
+import Card from '../../components/card';
+import { getUsers } from '../../service/apiClient';
+import './style.css';
+
+const AllSearchResults = () => {
+  const location = useLocation();
+  const { results: initialResults, searchVal: initialSearchVal } = location.state || { results: [], searchVal: '' };
+  const [searchVal, setSearchVal] = useState(initialSearchVal);
+  const [results, setResults] = useState(initialResults);
+  const [cohorts, setCohorts] = useState([]);   
+
+  useEffect(() => {
+      getUsers().then(setCohorts);
+      window.scrollTo(0, 0);
+  }, []);
+
+  const onChange = (e) => {
+    setSearchVal(e.target.value);
+    const filteredResults = cohorts.filter((cohort) => {
+      if (cohort.firstName && cohort.lastName) {
+        const fullName = `${cohort.firstName || ''} ${cohort.lastName || ''}`.toLowerCase();
```

Comment:
You don't need the `|| ''` here because you're inside a if statement that just checked that `cohort.firstName` and `cohort.lastName` are not falsy.

Also, the naming is a bit weird here - how does a cohort have a first and last name?